tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -458,6 +459,81 @@ where
         }
     }
 
+    fn on_enter(&self, id: &Id, ctx: Context<'_, S>) {
+        let span = ctx.span(id).expect("Span not found, this is a bug");
+        let mut extensions = span.extensions_mut();
+        if let Some(timings) = extensions.get_mut::<Timings>() {
+            timings.entered = Some(Instant::now());
+        } else {
+            extensions.insert(Timings::new());
+        }
+    }
+
+    fn on_exit(&self, id: &Id, ctx: Context<'_, S>) {
+        let span = ctx.span(id).expect("Span not found, this is a bug");
+        let mut extensions = span.extensions_mut();
+        if let Some(timings) = extensions.get_mut::<Timings>() {
+            if let Some(started) = timings.entered.take() {
+                timings.accumulated += started.elapsed().as_nanos() as u64;
+            }
+        }
```

Comment:
I think that if we merge this, we should probably allow users to configure whether or not we record these timings --- the additional timestamps can be expensive. There should probably be a builder method to configure whether or not we emit span close events (I'm not sure whether allowing span durations to be disabled separately is necessary or not --- it might be too granular, and I dunno if there's much demand for the span end events without the durations?)

Reply:
Yes, I’d like to offer the following choices:

- ignore them (just like now)
- print one event on creation and one on exit
- print one event on exit with the timings

WDYT? Or we break it up into three individual options:

- print creation
- print exit
- include timing

Reply:
Hmm, my initial thought is that having toggles for logging an event in `new_span` and in `on_close`, and always emitting the timings if close events are enabled is _probably_ enough to start with --- we can add a separate switch for configuring the timings later if people ask for it.

Another thought is that it might be good if the `without_time` method, which disables the timestamps on each log line also disabled timing spans. On the other hand, the span timings may actually be lighter-weight than timestamps for every event, since they only require `Instant`s rather than `SystemTime`s (so they may not always require a syscall), and they may not be required as frequently, depending on the number of spans and events in the code being traced. So, maybe having them configurable separately _is_ a good idea. I'm open to being convinced either way. What do you think? 

@davidbarsky or @LucioFranco may have opinions on what the best API is, as well.